src/commands/deployments/editScmType.ts

```diff
@@ -13,6 +13,9 @@ export async function editScmType(actionContext: IActionContext, node?: SiteTree
     if (!node) {
         node = <DeploymentsTreeItem>await ext.tree.showTreeItemPicker([DeploymentsTreeItem.contextValueConnected, DeploymentsTreeItem.contextValueUnconnected]);
     } else if (node instanceof SiteTreeItem) {
+        if (!node.deploymentsNode) {
```

Comment:
I think a better fix would be to define `deploymentsNode` in the constructor of `SiteTreeItem` (you don't even have to check the tier, you could just default it to something). Otherwise it feels like the type of `deploymentsNode` should be changed to `| undefined` so that strictNullChecks will find any other similar bugs

Reply:
With changing the constructor, but unless you refresh the web app, you won't have an accurate ScmType on the Deployments node.  I could make `loadMoreChildren` update `this.deploymentsNode` immediately but that feels kind of pointless.

The type actually is defined as `DeploymentsTreeItem | undefined` but I cast it as `<DeploymentsTreeItem>` which is why I missed this error 😓